containers/ecr-viewer/src/app/api/services/database_repo.ts

```diff
@@ -0,0 +1,360 @@
+import { Kysely } from "kysely";
+
+import { db } from "./database";
+import {
+  ECRConditions,
+  NewECRConditions,
+  ECRConditionsUpdate,
+  ECRRuleSummaries,
+  NewECRRuleSummaries,
+  ECRRuleSummariesUpdate,
+  ECR,
+  NewECR,
+  ECRUpdate,
+} from "./types";
+import { Core } from "./types";
+
+// ECR_DATA
+
+/**
+ * Finds an eICR by its ID
+ * @async
+ * @function findEcrById
+ * @param id - thd ID of the eCR to be looked up
+ * @returns an eICR object
+ */
+export async function findEcrById(id: string | null) {
+  if (!id) {
+    return console.error("eICR ID is required.");
+  }
+  try {
+    return await (db as Kysely<Core>)
+      .selectFrom("ecr_data")
+      .where("eICR_ID", "=", id)
+      .selectAll()
+      .executeTakeFirst();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**
+ * Finds an eICR by its criteria
+ * @async
+ * @function findEcr
+ * @param criteria - the Partial<ECR> used to filter against
+ * @returns an eICR object
+ */
+export async function findEcr(criteria: Partial<ECR> | null) {
+  let query = (db as Kysely<Core>).selectFrom("ecr_data");
+
+  if (!criteria || criteria === null) {
+    return console.error("eICR Criteria is required.");
+  }
+
+  if (criteria.eICR_ID) {
```

Comment:
it seems like for most of the criteria, we're following a very standard pattern, could we do something like
```js
for (const criterium of ["eICR_ID", "set_id", ...]) {
  if (criteria[criterium]) {
    query = query.where(criterium, "=", criteria[criterium])
  }
}
```

Reply:
we could maybe fold in the `null` handling for all of them and have it that way?

Or even just iterate over all the keys in criteria and keep this totaly flexible instead of set on specific keys?